src/vs/platform/files/common/files.ts

```diff
@@ -105,6 +105,27 @@ export interface IFileService {
 	del(resource: URI, useTrash?: boolean): TPromise<void>;
 
 	/**
+	 * Backs up the provided file to a temporary directory to be used by the hot
+	 * exit feature and crash recovery.
+	 */
+	backupFile(resource: URI, content: string): TPromise<IFileStat>;
```

Comment:
For hot exit we care about untitled files as well, I put this in `IFileService` as per your advice due to this?